lms/static/js/spec/instructor_dashboard/data_download_spec.js

```diff
@@ -0,0 +1,74 @@
+
+            it('show data on success callback', function() {
+                // Spy on AJAX requests
+                var requests = AjaxHelpers.requests(this);
+                data_download_certificate.$list_issued_certificate_table_btn.data('endpoint', url);
+                var data = {
+                    'certificates': [{'course_id':'xyz_test', 'mode':'honor'}],
+                    'queried_features': ['course_id', 'mode'],
+                    'feature_names': { 'course_id': 'Course ID', 'mode': ' Mode'}
+                };
+
+                data_download_certificate.$list_issued_certificate_table_btn.click();
+                AjaxHelpers.expectJsonRequest(requests, 'POST', url);
+
+                // Simulate a success response from the server
+                AjaxHelpers.respondWithJson(requests, data);
+                expect(data_download_certificate.$certificate_display_table.html()
+                    .indexOf('Course ID') !== -1).toBe(true);
+                expect(data_download_certificate.$certificate_display_table.html()
+                    .indexOf('Mode') !== -1).toBe(true);
+                expect(data_download_certificate.$certificate_display_table.html()
+                    .indexOf('xyz_test') !== -1).toBe(true);
+                expect(data_download_certificate.$certificate_display_table.html()
+                    .indexOf('honor') !== -1).toBe(true);
+            });
+
+            it('show error on failure callback', function() {
+                // Spy on AJAX requests
+                var requests = AjaxHelpers.requests(this);
+
+                data_download_certificate.$list_issued_certificate_table_btn.click();
+                // Simulate a error response from the server
+                AjaxHelpers.respondWithError(requests);
+                expect(data_download_certificate.$certificates_request_response_error.text())
+                    .toEqual('Error getting issued certificates list.');
+            });
+
+            it('error should be clear from UI on success callback', function() {
+                var requests = AjaxHelpers.requests(this);
+                data_download_certificate.$list_issued_certificate_table_btn.click();
+
+                // Simulate a error response from the server
+                AjaxHelpers.respondWithError(requests);
+                expect(data_download_certificate.$certificates_request_response_error.text())
+                    .toEqual('Error getting issued certificates list.');
+
+                // Simulate a success response from the server
+                data_download_certificate.$list_issued_certificate_table_btn.click();
+                AjaxHelpers.expectJsonRequest(requests, 'POST', url);
+
+                expect(data_download_certificate.$certificates_request_response_error.text())
+                    .not.toEqual('Error getting issued certificates list');
+            });
+
+            it('data list button click callback', function() {
```

Comment:
This test does not make sense. We are triggering a `click` event in one statement and in the next statement we are expecting `click` event to be triggered. We should probably check the event handler is called or not.